core/services/beacon/settings.py

```diff
@@ -152,3 +152,28 @@ def migrate(self, data: Dict[str, Any]) -> None:
             logger.error(f"unable to update SettingsV1 to SettingsV2: {e}")
 
         data["VERSION"] = SettingsV2.VERSION
+
+
+class SettingsV3(SettingsV2):
```

Comment:
This should use V1 since V2 does not add extra fields. 
```suggestion
class SettingsV3(SettingsV1):
```

Reply:
Wait. Won't this jump V2 migration?

Reply:
Check my next comment.

Reply:
Got it. This is kinda hacky thought. In the end it is a bug that we are working around. Did you find out why this happens, or a fix in the library level? Writing the settings is a lot of work already. This problem will surely happen again.

Reply:
I have not gone deeply on the issue, I believe that the logic behind `extra_attributes` or `__get_fields_*` is failing to do the proper job, from deriving of a class without extra attributes deriving from a class that has no extra attributes while having a super.super class that contains `JsonObject`.
Since we are aiming to release a stable version of BlueOS and hotspot is a key feature, I would recommend the workaround of not deriving from classes that does not contains extra fields for now.
I'll investigate it a bit more.